doc/user/content/manage/troubleshooting.md

```diff
@@ -183,7 +183,66 @@ ORDER BY elapsed_ns DESC
+
+If you’re planning to run a query regularly and expect it to be fast, an index is highly recommended.
+
+Some tips for troubleshooting using an index:
+
+- Start by querying the [EXPLAIN](https://materialize.com/docs/sql/explain/) plan. This will tell you the query plan that Materialize will execute for your query.
+- Once you add an index, re-issue the explain plan to ensure the index was actually used.
+- If you have already created an index, and it’s not showing up in the explain plan, you may be running the query in a different cluster than where the index resides. An index caches the results in memory on the cluster on which you create it. You need to execute your query on that same cluster in order to utilize the index.
+
+If you do not want to use an index, consider the size of the underlying dataset on which you’re querying. Without an index, the query is computed on the fly, so larger data sizes translate to more on-the-fly computation, which can lead to longer latencies.
+
+For more on when to use an index, check out our blog post [When to Use Indexes and Materialized Views](https://materialize.com/blog/views-indexes/).
+
+### Isolation level
+
+Isolation level is a way to specify how you want your query to behave with respect to the other queries, source ingestion, and index & materialize view updates happening at the same time within your Materialize region. All isolation levels at Materialize guarantee you’re getting a consistent snapshot of the data - we provide the effect that all transactions are running in serial order and you will not see partially-committed transaction data.
+
+The default isolation level is Strict Serializable, which ensures that for each query you execute, you will always see results that are at least as fresh as previous runs of the query. You get the most up to date correct result each time. The tradeoff with Strict Serializable is that you may need need to wait for data ingested and updates made at a previous time to propagate through materialized views and indexes before you can run your next query, resulting in your queries being occasionally blocked.
+
+An alternative isolation level is Serializable. With Serializable level, the results are still guaranteed to be correct, but may be from an earlier point in time. As a result, queries do not have to wait for recent updates and data ingestion to propagate, thus reducing the likelihood of your query being blocked. In Serializable, queries that involve a single object should never block.
+
+What you can do: if you...
+
+- care more about latency than freshness, and would prefer to receive fast, stale results \
+AND
+- are ok with queries "traveling back in time" with respect to other queries in order to serve those fast, stale results
+
+then you can switch your isolation level to Serializable.
+
+To change your session's isolation level to Serializable, run
+```
+SET transaction_isolation TO 'SERIALIZABLE';
```

Comment:
Todo: add the below once default session vars per role is in GA

> To make this the value the default for your role, so that it applies every time you connect, run
> 
> `ALTER ROLE <your role> SET transaction_isolation = 'SERIALIZABLE';`